packages/webdriverio/src/commands/constant.js

```diff
@@ -91,3 +91,17 @@ export const touchAction = function (actions) {
     formattedAction.forEach((params) => validateParameters(params))
     return protocolCommand(formattedAction)
 }
+
+// generate a function that can be used to query shadowRoots
+export const shadowFnFactory = function(elementSelector, qsAll) {
+    const strFn = `
+    (function() {
+      // element has a shadowRoot property
+      if (this.shadowRoot) {
+        return this.shadowRoot.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+      }
+      // fall back to querying the element directly if not
+      return this.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+    })`
```

Comment:
Let's move this to https://github.com/webdriverio/webdriverio/tree/master/packages/webdriverio/src/scripts. I don't thing we need this helper then anymore.